### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by meta-nilesh-patel**
Only check assignment 4!

### CODE REVIEW

File: SCF/Assignment4/ArrOperation.java [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+	/**
+	 * 
+	 * For validating input array is not empty and contain only positive integer
+	 * @param inputArray positive integer array
+	 * @return true if array contain only positive integer
+	 *         false if array contain negative integer
+	 */
+	private boolean inputValidation(int inputArray[])
+	{
+
```
Comment (rash-tiwari) [inline SCF/Assignment4/ArrOperation.java]: Correct the formatting for this method

File: SCF/Assignment4/TestMethodFixXYInArray.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+	private int[] inputArray;
+	private int x;
+	private int y;
+	private int[] expectedOutput;
+	static ArrOperation objectOfArrOperationCalss = new ArrOperation();
+	
+	public TestMethodFixXYInArray(int[] input, int x, int y, int[] expectedOutput)
+	{
+			this.inputArray = input;
```
Comment (rash-tiwari) [inline SCF/Assignment4/TestMethodFixXYInArray.java]: Remove extra tab from here.

File: SCF/Assignment4/TestMethodGetSplitIndexInArray.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+public class TestMethodGetSplitIndexInArray 
+{
+
+	private int[] inputArray;
+	private Object expectedOutput;
+	static ArrOperation objectOfArrOperationCalss = new ArrOperation();
+	
+	public TestMethodGetSplitIndexInArray(int[] input, Object expectedOutput)
+	{
+			this.inputArray = input;
```
Comment (rash-tiwari) [inline SCF/Assignment4/TestMethodGetSplitIndexInArray.java]: remove unnecessary tab

File: SCF/Assignment4/ArrOperation.java [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+	 * @throws AssertionError when array contain negative number
+	 */
+	public int getNumberOfClumpsInArray(int inputArray[]) throws AssertionError
+	{
+
+		//input validation
+		if(inputArray.length == 0)
+		{
+			throw new AssertionError("Null Array");
+		}else
```
Comment (rash-tiwari) [inline SCF/Assignment4/ArrOperation.java]: Add else in new line

**Review by rash-tiwari — CHANGES_REQUESTED** [review]
1st Level Code Review Done

File: SCF/Assignment4/ArrOperation.java (hunk 1 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment4/ArrOperation.java]: Done

File: SCF/Assignment4/TestMethodFixXYInArray.java (hunk 2 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment4/TestMethodFixXYInArray.java]: Done

File: SCF/Assignment4/TestMethodGetSplitIndexInArray.java (hunk 3 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment4/TestMethodGetSplitIndexInArray.java]: Done

File: SCF/Assignment4/ArrOperation.java (hunk 4 above)
Comment (meta-nilesh-patel) [reply SCF/Assignment4/ArrOperation.java]: Done

**Review by rash-tiwari — APPROVED** [review]

**Pull request merged**